Framework/Algorithms/src/Q1D2.cpp

```diff
@@ -146,7 +146,7 @@ void Q1D2::exec() {
   Progress progress(this, 0.05, 1.0, numSpec + 1);
 
   const auto &spectrumInfo = m_dataWS->spectrumInfo();
-  PARALLEL_FOR3(m_dataWS, outputWS, pixelAdj)
+  PARALLEL_FOR_IF(Kernel::threadSafe(*m_dataWS, *outputWS, pixelAdj.get()))
```

Comment:
No, `pixelAdj` can be `NULL`.  I tried dereferencing it and the unit test failed.